tools/castle-editor/code/framedesign.pas

```diff
@@ -1857,6 +1863,29 @@ procedure TDesignFrame.ControlsTreeDragOver(Sender, Source: TObject; X,
   ControlsTree.Invalidate; // force custom-drawn look redraw
 end;
 
+procedure TDesignFrame.ControlsTreeEditing(Sender: TObject; Node: TTreeNode;
+  var AllowEdit: Boolean);
+begin
+  { this event is fired when calling TCustomListView.CanEdit which itself is called in TCustomListView.ShowEditor
+    therefore this event preceeds initializing and showing of the editor }
+  //here we have to "restore" the pure name of the component (without class name) before starting edit
+  Node.Text := TComponent(ControlsTree.Selected.Data).name;
+end;
+
+procedure TDesignFrame.ControlsTreeEditingEnd(Sender: TObject; Node: TTreeNode;
+  Cancel: Boolean);
+var
+  S: String;
+  Sel: TComponent;
+begin
+  Sel := TComponent(ControlsTree.Selected.Data);
```

Comment:
Similar comment as the previous:

- Can we use `TComponent(Node.Data)` instead of `TComponent(ControlsTree.Selected.Data)`?

- We prefer CamelCase, so write `Sel.Name` instead of `Sel.name` in the line below.

Reply:
Done in 112b252dba6a348ee86456ee5ec06bd8383b68dc